tckdb/backend/app/tests/endpoints/conftest.py

```diff
@@ -0,0 +1,214 @@
+    ALEMBIC_INI = BASE_DIR / "alembic.ini"
+    print(BASE_DIR)
+    # Ensure the alembic.ini file exists
+    assert ALEMBIC_INI.exists(), f"Alembic config file not found at {ALEMBIC_INI}"
+
+    # Configure Alembic
+    alembic_cfg = Config(str(ALEMBIC_INI))
+    alembic_cfg.set_main_option(
+        "sqlalchemy.url", SQLALCHEMY_DATABASE_URL
+    )  # Override URL
+
+    # Run migrations
+    try:
+        command.upgrade(alembic_cfg, "head")
+        print("Alembic migrations applied.")
+    except Exception as e:
+        print(f"Error applying migrations: {e}")
+        raise
+
+    # **Debugging: Check if 'level' table exists**
+    try:
+        with engine.connect() as connection:
+            statement = text("SELECT to_regclass('public.level');")
+            result = connection.execute(statement)
+            table_exists = result.fetchone()[0] is not None
+            if table_exists:
+                print("Table 'level' exists in the database.")
+            else:
+                print("Table 'level' does NOT exist in the database.")
+                raise Exception("Migration did not create 'level' table.")
+    except Exception as e:
+        print(f"Error during migration verification: {e}")
+        raise
+
+    yield
+
+    # Downgrade migrations after tests
+    try:
+        command.downgrade(alembic_cfg, "base")
+        print("Alembic migrations downgraded.")
+    except Exception as e:
+        print(f"Error downgrading migrations: {e}")
+        raise
+
+
+@pytest.fixture(scope="class")
+def db_session(setup_database):
+    """Create a new database session for a test module."""
+    connection = engine.connect()
+    transaction = connection.begin()
+    session = TestingSessionLocal(bind=connection)
+    yield session
+    session.close()
+    if transaction.is_active:
+        transaction.rollback()
+    connection.close()
+
+
+@pytest.fixture(scope="class")
+def test_level(db_session):
+    """Create a temporary Level entry."""
+    level = Level(method="B3LYP", basis="6-31G(d,p)", dispersion="gd3bj")
+    db_session.add(level)
+    db_session.commit()
+    db_session.refresh(level)
+    return level
+
+
+@pytest.fixture(scope="class")
+def test_ess(db_session):
+    """Create a temporary ESS entry."""
+    ess = ESS(
+        name="TestESS", version="1.0", revision="A", url="http://testess.example.com"
+    )
+    db_session.add(ess)
+    db_session.commit()
+    db_session.refresh(ess)
+    return ess
+
+
+@pytest.fixture(scope="class")
+def test_encorr(db_session, test_level: Level):
+    """
+    Fixture to create a test EnCorr record.
+    """
+    encorr = EnCorr(
+        level_id=test_level.id,
+        supported_elements=["H", "C", "N", "O", "S"],
+        energy_unit="Hartree",
+        aec={
+            "H": -0.499459,
+            "C": -37.786694,
+            "N": -54.524279,
+            "O": -74.992097,
+            "S": -397.648733,
+        },
+        bac={
```

Comment:
all this hard coded data should move to `constants.py` file

Reply:
It's removed cause it isn't needed since the API changes. Thanks for pointing it out!